### PR TITLE
Make the Python binding of FGPropertyNode compatible with SGSharedPtr ref counting

### PR DESCRIPTION
**Pull request by bcoconni**
This PR modifies the Python binding of `FGPropertyNode` so that the reference counting of the `SGPropertyNode` instance that it points to is updated.

This is to avoid that the C++ instance of `FGPropertyNode`/`SGPropertyNode` that the Python object relies on is being deleted by another event. Such a scenario would result in a dangling pointer and eventually to a segfault of the Python interpreter.

In other words, this PR is meant to avoid the scenario below from occurring.

```python
fdm = jsbsim.FGFDMExec()
pm = fdm.get_property_manager()
node = pm.get_node('some/node')

# [...] A lot of code

del fdm

# [...] Some code until the Python garbage collector actually deletes fdm

node.get_double_value()  # BOOOMM !!!
```

### CODE REVIEW

**Comment by agodemar** [comment]
Thanks @bcoconni, well done

**Comment by seanmcleod** [comment]
@bcoconni @agodemar quite a milestone, the 1000th pull request for the JSBSim repo at Github! :wink:

**Comment by bcoconni** [comment]
It's actually the [298th-ish PR](https://github.com/JSBSim-Team/jsbsim/pulls): the number 1000 is actually a counter that accumulates over all the items in the repo (PR, issues and discussions).

That's quite a feat nonetheless. Congrats to the community 🥳
